modules/catalog/src/main/java/org/apache/ignite/internal/catalog/systemviews/ParentIdAwareDescriptor.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.catalog.systemviews;
+
+/**
+ * A container that keeps given descriptor along with its parent's id.
+ */
+final class ParentIdAwareDescriptor<T> {
```

Comment:
Personally, I would prefer to make such single-use containers (ParentIdAwareDescriptor, CatalogAwareDescriptor, SchemaAwareDescriptor) as private static nested classes. At least because they don’t have very obvious names.
But it's up to you.